webapp/portlet/src/main/webapp/vue-apps/common/initComponents.js

```diff
@@ -189,4 +191,50 @@ Vue.directive('cacheable', {
       vnode.componentInstance.$root.$emit('application-mount', true);
     }, 10000);
   },
-});
\ No newline at end of file
+});
+
+Vue.directive('draggable', {
+  bind(el, binding) {
+    let counter = 0;
+    const dropZoneBindingValue = binding?.value;
+    const dropZoneParentElement = dropZoneBindingValue?.parentElementId ? document.getElementById(dropZoneBindingValue.parentElementId) : el;
+    const attachmentEnabled = dropZoneBindingValue?.attachmentEnabled;
+    if (attachmentEnabled) {
+      ['drag', 'dragstart', 'dragend', 'dragover', 'dragenter', 'dragleave', 'drop'].forEach((event) => {
+        dropZoneParentElement.addEventListener(event, (e) => {
+          if (e?.dataTransfer) {
+            e.preventDefault();
+            e.stopPropagation();
+          }
+        });
+      });
+      ['dragenter', 'dragstart'].forEach((event) => {
+        dropZoneParentElement.addEventListener(event, (e) => {
+          if (e?.dataTransfer) {
+            counter++;
+            document.dispatchEvent(new CustomEvent('attach-image-show-drop-zone'));
+          }
+        });
+      });
+      ['dragleave', 'dragend'].forEach((event) => {
+        dropZoneParentElement.addEventListener(event, (e) => {
+          if (e?.dataTransfer) {
+            counter--;
+            if (counter === 0) {
+              document.dispatchEvent(new CustomEvent('attach-image-hide-drop-zone'));
```

Comment:
This directive could be used for other type of attachments, not only image attachments.

```suggestion
              document.dispatchEvent(new CustomEvent('attachments-hide-drop-zone'));
```

Reply:
done